pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -53,19 +53,19 @@ type Registry struct {
 	Prefix string `toml:"prefix"`
 }
 
-// backwards compatability to sysregistries v1
-type v1TOMLregistries struct {
+// V1TOMLregistries is for backwards compatability to sysregistries v1
+type V1TOMLregistries struct {
 	Registries []string `toml:"registries"`
 }
 
-// tomlConfig is the data type used to unmarshal the toml config.
-type tomlConfig struct {
+// TomlConfig is the data type used to unmarshal the toml config.
+type TomlConfig struct {
 	Registries []Registry `toml:"registry"`
 	// backwards compatability to sysregistries v1
 	V1Registries struct {
```

Comment:
I don’t think this structure, which combines both v1 and v2, and is really an implementation detail of how this package does parsing, should be made public.

If only the v1 structure is necessary, would it be possible to make a public `V1TOMLConfig` which contains only the `V1Registries` field (possibly renamed named `Registries`), and embed that `V1TOMLConfig` as an unnamed field within (private) `tomlConfig`?

Reply:
Did you want public only the definition of the `V1TOMLRegistries` element, or all of `V1Registries` with the top-level `search`/`insecure`/`block` fields?

Reply:
Yup, just did that. I followed what you sort of described in the comment above. Let me know if I missed something.